File: financeiro-api/src/main/java/br/com/iftm/financeiroapi/model/domain/Entry.java
package br.com.iftm.financeiroapi.model.domain;

import br.com.iftm.financeiroapi.model.utils.IdentifierUtil;
import com.fasterxml.jackson.annotation.JsonFormat;
import io.swagger.annotations.ApiModelProperty;

import java.math.BigDecimal;
import java.util.Date;
import java.util.HashSet;
import java.util.Set;

public class Entry {

    @ApiModelProperty(hidden = true)
    private String id;

    private String description;

    @JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "yyyy-MM-dd HH:mm:ss")
    private Date date;

    private BigDecimal value;

    private Set<Category> categories;

    public Entry() {
        this.id = IdentifierUtil.generateUUID();
    }

    public String getId() {
        return id;
    }

    public void setId(String id) {
        this.id = id;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public Date getDate() {
        return date;
    }

    public void setDate(Date date) {
        this.date = date;
    }

    public BigDecimal getValue() {
        return value;
    }

    public void setValue(BigDecimal value) {
        this.value = value;
    }

    public Set<Category> getCategories() {
        if(categories == null) {
            categories = new HashSet<>();
        }
        return categories;
    }

    public void setCategories(Set<Category> categories) {
        this.categories = categories;
    }

}

File: financeiro-api-kotlin/src/main/kotlin/br/com/iftm/financeiroapi/model/domain/Entry.kt
package br.com.iftm.financeiroapi.model.domain

import br.com.iftm.financeiroapi.model.extends.generateUUID
import com.fasterxml.jackson.annotation.JsonFormat
import io.swagger.annotations.ApiModelProperty
import java.math.BigDecimal
import java.util.*


data class Entry (
        var description: String = "",
        @JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "yyyy-MM-dd HH:mm:ss")
        var date: Date = Date(),
        var value: BigDecimal = BigDecimal.ZERO,
        var categories: Set<Category> = emptySet()
) {
    @ApiModelProperty(hidden = true)
    var id: String = generateUUID()
}
